### ISSUE TITLE
 [Dev 3.3.0-dev-e] Nightscout synchronization issue missing from prefs

### ISSUE THREAD

**Issue by emilisev** [nightscout/AndroidAPS#3569]
Seen Dev 3.3.0-dev-e, also dev c and d (I don't know for prior versions, this is my first time building dev).
The "Synchonization" line in Nightscout prefs is missing, or rather disappears at once when displaying the page. It's the same for NS v1 or v3. Happens in all circumstances, with a fresh AAPS install and no parameters whatsoever, or with imported parameters and NS sync working, both v1 and v3.
Tested on 2 different phones, Fairphone 5, Honor, and also Android Emulator right from AS, on 2 different computers...
Long story short : happens everywhere and everytime.

![Screenshot_20241124-1343592](https://github.com/user-attachments/assets/6f6fcf93-6a1b-44aa-a9a1-1c893ac205ca)

https://github.com/user-attachments/assets/5d5dbace-3c43-411b-a0b0-f10a70b83b30

[AndroidAPS.log](https://github.com/user-attachments/files/17968732/AndroidAPS.log)

**Comment by MilosKozak** [comment]
I have it here. Are you sure you are bulding full AAPS and not AAPSClient?

**Comment by emilisev** [comment]
Yes this is the fullRelease. AAPSClient doesn't behave the same, the line is completely absent, not present but then disappears.

**Comment by MilosKozak** [comment]
how can you use dev version without engeneering mode?

**Comment by emilisev** [comment]
I didn't know this was mandatory. I'm not using dev version yet - nor planning to, actually I just wanted to test my changes about the documentation URL changes.
I added the file and it solved my issue. 

It is not clear from the doc that it is mandatory to use engeneering in dev.
https://androidaps.readthedocs.io/en/latest/AdvancedOptions/DevBranch.html
"Therefore many unfinished features are disabled. To enable these features enter Engineering Mode by creating a file named engineering_mode in directory /AAPS/extra . Enabling the engineering mode might break the loop entirely."

Should I update the doc to make it clear that it is mandatory ?

**Comment by robertrub** [comment]
Last dev

![Screenshot_20241201_132514_AAPS.jpg](https://github.com/user-attachments/assets/036fbb68-ab58-4b8e-b2dc-598bd95acc61)

Connect to power so that the upload to NS triggers. Data uploaded normally but then stopped with 38 in queue. Phone was still connected to power.

![Screenshot_20241201_132606_AAPS.jpg](https://github.com/user-attachments/assets/08a90b11-7ba5-4bf9-a593-bd35c0431179)

The logs. Look around 13:25

![AndroidAPS_LOG_1733056070119.log.zip](https://github.com/user-attachments/files/17968903/AndroidAPS_LOG_1733056070119.log.zip)

**Comment by MilosKozak** [comment]
fixed in dev

**Comment by MilosKozak** [comment]
https://github.com/nightscout/AndroidAPS/commit/42909b14046985d869616108776dc39a6dc49331
https://github.com/nightscout/AndroidAPS/commit/0f89dfd159f235377dfc71b92acdc28fe9e8b3e8

**Comment by vanelsberg** [comment]
>"Therefore many unfinished features are disabled. To enable these features enter Engineering Mode by creating a file named >engineering_mode in directory /AAPS/extra . Enabling the engineering mode might break the loop entirely."

> **Should I update the doc to make it clear that it is mandatory ?**

@emilisev 
Even though most dev do understand and AAPS is complaining loudly on not running dev mode I agree this is not clear for non-devs. Text above suggest "unfinished features" only but actually many things will not work properly.

So yes, I would think doc should be updated: settings Engineering Mode is mandatory for running dev builds/
